Sources/TuistKit/Services/Inspect/InspectBuildService.swift

```diff
@@ -0,0 +1,197 @@
+
+    var errorDescription: String? {
+        switch self {
+        case let .projectNotFound(path):
+            return "No Xcode project found at \(path.pathString). Make sure it exists."
+        case let .noBuildLogFound(buildLogsPath: buildLogsPath, projectPath: projectPath):
+            return "No build logs for project \(projectPath.basename) found at \(buildLogsPath.pathString)."
+        case .missingFullHandle:
+            return "The 'Tuist.swift' file is missing a fullHandle. See how to set up a Tuist project at: https://docs.tuist.dev/en/server/introduction/accounts-and-projects#projects"
+        case .executablePathMissing:
+            return "We couldn't find tuist's executable path to run inspect build in a background."
+        }
+    }
+}
+
+struct InspectBuildService {
+    private let environment: Environmenting
+    private let derivedDataLocator: DerivedDataLocating
+    private let fileSystem: FileSysteming
+    private let ciChecker: CIChecking
+    private let machineEnvironment: MachineEnvironmentRetrieving
+    private let xcodeBuildController: XcodeBuildControlling
+    private let createBuildService: CreateBuildServicing
+    private let configLoader: ConfigLoading
+    private let xcactivityParser: XCActivityParsing
+    private let backgroundProcessRunner: BackgroundProcessRunning
+    private let dateService: DateServicing
+
+    init(
+        environment: Environmenting = Environment.shared,
+        derivedDataLocator: DerivedDataLocating = DerivedDataLocator(),
+        fileSystem: FileSysteming = FileSystem(),
+        ciChecker: CIChecking = CIChecker(),
+        machineEnvironment: MachineEnvironmentRetrieving = MachineEnvironment.shared,
+        xcodeBuildController: XcodeBuildControlling = XcodeBuildController(),
+        createBuildService: CreateBuildServicing = CreateBuildService(),
+        configLoader: ConfigLoading = ConfigLoader(),
+        xcactivityParser: XCActivityParsing = XCActivityParser(),
+        backgroundProcessRunner: BackgroundProcessRunning = BackgroundProcessRunner(),
+        dateService: DateServicing = DateService()
+    ) {
+        self.environment = environment
+        self.derivedDataLocator = derivedDataLocator
+        self.fileSystem = fileSystem
+        self.ciChecker = ciChecker
+        self.machineEnvironment = machineEnvironment
+        self.xcodeBuildController = xcodeBuildController
+        self.createBuildService = createBuildService
+        self.configLoader = configLoader
+        self.xcactivityParser = xcactivityParser
+        self.backgroundProcessRunner = backgroundProcessRunner
+        self.dateService = dateService
+    }
+
+    func run(
+        path: String?
+    ) async throws {
+        let referenceDate = dateService.now()
+        guard let executablePath = Bundle.main.executablePath else { throw InspectBuildServiceError.executablePathMissing }
+
+        if environment.tuistVariables["TUIST_INSPECT_BUILD_WAIT"] != "YES",
```

Comment:
I don't remember if we have an `isTruthy` in this codebase, but I'd suggest to use something like that to work in cases where people assumes that `1` and `true` are also valid values.

Reply:
I'd do the same with all the references in this file.

Reply:
In that case it's fine.